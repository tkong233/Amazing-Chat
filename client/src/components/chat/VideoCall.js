import React, { useState, useCallback, useEffect } from "react";
import { connect } from "react-redux";
import Button from "@material-ui/core/Button";
import Room from "./Room";
import axios from "axios";

import { hangUpVideoCall, sendMessage } from "../../actions/chatActions";

const VideoCall = (props) => {
  const { pairId, socket, sender, receiver } = props.chat;
  // console.log(`chatprops: ${props.chat}`);
  // console.log(props.user);
  const [username, setUsername] = useState("");
  const [roomName, setRoomName] = useState("");
  const [token, setToken] = useState(null);

  useEffect(() => {
    setUsername(props.user.name);
    setRoomName(pairId);
    axios
      .post("/video/token", {
        identity: props.user.name,
        room: pairId,
      })
      .then((data) => {
        // console.log(data.data.token);
        setToken(data.data.token);
      })
      .catch((err) => {
        console.log(err);
      });
  }, []);

  const handleLogout = useCallback((timeElapsed) => {
    setToken(null);
    const seconds = Math.floor(timeElapsed / 1000);
    const minutes = Math.floor((timeElapsed / 1000) / 60);
    const duration = minutes + " minutes " + seconds + " seconds "

    const message = "Video Call Duration: " + duration;
    const data1 = {
      pairId,
      from: sender,
      to: receiver,
      message,
      type: "text",
    };
    // console.log(data1);
    // console.log(socket);
    props.sendMessage(data1, socket);

    props.hangUpVideoCall(socket, pairId, sender, receiver);
  }, []);

  // set

  return (
    <div style={{ marginLeft: "3%", marginRight: "3%" }} data-test='VideoCallComponent'>
      {/* <h3>This is video call view</h3> */}
      {/* <h3>{token}</h3> */}
      {/* <h3>{username}</h3> */}
      {/* <h3>{roomName}</h3> */}
      {/* <Button
        variant="contained"
        color="secondary"
        onClick={() => {
          props.hangUpVideoCall(socket, pairId, sender, receiver);
        }}
      >
        Hang Up
      </Button> */}
      {token && (
        <div className="container">
          <Room roomName={roomName} token={token} handleLogout={handleLogout} />
        </div>
      )}
    </div>
  );
};

const mapStateToProps = (state) => ({
  user: state.auth.user,
  chat: state.chat,
});

export default connect(mapStateToProps, {
  hangUpVideoCall, sendMessage
})(VideoCall);
