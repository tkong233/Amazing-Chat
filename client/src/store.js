import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import { persistStore, persistReducer } from 'redux-persist'
import rootReducer from "./reducers";
import storage from 'redux-persist/lib/storage' // defaults to localStorage for web

const initialState = {};

const persistConfig = {
  key: 'root',
  storage,
}

const middleware = [thunk];

const persistedReducer = persistReducer(persistConfig, rootReducer, initialState);

const store = createStore(

  persistedReducer,
  compose(
    applyMiddleware(...middleware),
    (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__()) ||
      compose
  )
);

const persistor = persistStore(store);

export {
  store,
  persistor,
  middleware
};



