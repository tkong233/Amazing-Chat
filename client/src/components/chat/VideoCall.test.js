import React from 'react';
import VideoCall from './VideoCall';
import {shallow} from 'enzyme';
import checkPropTypes from 'check-prop-types';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16'; 
import { applyMiddleware, createStore } from 'redux';
import {middleware} from '../../store';
import rootReducer from '../../reducers/index'

Enzyme.configure({
    adapter: new EnzymeAdapter(),
});

const setUp = (initialState={}) =>{
    const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
    const store = createStoreWithMiddleware(rootReducer, initialState);
    const wrapper = shallow(<VideoCall store={store} />).dive();
    // console.log(wrapper.debug());
    return wrapper;
}

describe('VideoCall Component', ()=>{
    describe('Checking PropTypes', ()=>{
        it('Should not throw a warning', ()=>{
            const expectedProps = {
                user: {},
                chat: {}
            }
            // eslint-disable-next-line react/forbid-foreign-prop-types
            const propsErr = checkPropTypes(VideoCall.propTypes, expectedProps, 'props', VideoCall.name);
            expect(propsErr).toBeUndefined();
        })
    });
    describe('Renders', ()=>{
        let wrapper;
        beforeEach(()=>{
            const initialState = {
                auth:{
                    isAuthenticated: false,
                    user: {},
                    loading: false
                  },
                chat:{
                    itemStatusMap:
                    { 1 : 'LOADED'},
                    messages:[{
                        pairId: '',
                        from: 'userA', 
                        to: 'userB',
                        message: 'test message',
                        datetime: '2020/09/01',
                        type: 'text'
                    }],
                    sender: 'userA',
                    receiver: 'userB'
                }
            };
            wrapper = setUp(initialState);
        });

        it('Should render without errors', ()=>{
            const container = wrapper.find(`[data-test='VideoCallComponent']`);
            expect(container.length).toBe(1);
        });

        it('Snapshot testing', ()=>{
            expect(wrapper).toMatchSnapshot();
        });


    });
});