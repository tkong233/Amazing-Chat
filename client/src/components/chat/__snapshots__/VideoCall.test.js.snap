// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoCall Component Renders Snapshot testing 1`] = `ShallowWrapper {}`;
